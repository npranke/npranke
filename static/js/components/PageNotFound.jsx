import Home from '@img/icon-home.png'

import {
    getLocationPageTitle,
    getSendEventHandler,
    sendPageview,
} from '@utils'

function PageNotFound() {
    document.title = getLocationPageTitle('pagenotfound')

    sendPageview()

    return (
        <main className="pagenotfound">
            <div className="intro">
                Oops! <br />
                Page not found.
            </div>
            <div className="button-home">
                <a
                    onClick={ getSendEventHandler(
                        'pagenotfound',
                        'navigate',
                        'home',
                    ) }
                    href="/home"
                    target="_self"
                    rel="noreferrer"
                >
                    <img
                        src={ Home }
                        className="icon icon-home"
                        alt="Home icon"
                    />
                    <span className="text-home">Home</span>
                </a>
            </div>
        </main>
    )
}

export default PageNotFound
