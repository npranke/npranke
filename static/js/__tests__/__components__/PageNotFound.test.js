import Adapter from 'enzyme-adapter-react-16'
import Enzyme, { shallow } from 'enzyme'
import React from 'react'
import { render } from '@testing-library/react'

import PageNotFound from '@components/PageNotFound'

import * as utils from '@utils'

Enzyme.configure({ adapter: new Adapter() })

describe('PageNotFound', () => {
    test('sets document title', () => {
        shallow(<PageNotFound />)

        expect(document.title).toContain('pagenotfound')
    })

    test('sends pageview', () => {
        utils.sendPageview = jest.fn()

        shallow(<PageNotFound />)

        expect(utils.sendPageview).toHaveBeenCalledTimes(1)
    })

    test('has intro text with oops', () => {
        const pageNotFound = shallow(<PageNotFound />)

        expect(
            pageNotFound.find('.intro').text(),
        ).toContain('Oops!')
    })

    test('has link to home', () => {
        const pageNotFound = shallow(<PageNotFound />)

        expect(
            pageNotFound.find('.button-home a').props().href,
        ).toEqual('/home')
    })

    test('has alt text for home icon', () => {
        const pageNotFound = shallow(<PageNotFound />)

        expect(
            pageNotFound.find('.button-home img').props().alt,
        ).toEqual('Home icon')
    })
})

describe('PageNotFound snapshot', () => {
    test('matches snapshot', () => {
        const { asFragment } = render(<PageNotFound />)

        expect(asFragment()).toMatchSnapshot()
    })
})
