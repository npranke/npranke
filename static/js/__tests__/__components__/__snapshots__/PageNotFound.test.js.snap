// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageNotFound snapshot matches snapshot 1`] = `
<DocumentFragment>
  <main
    class="pagenotfound"
  >
    <div
      class="intro"
    >
      Oops! 
      <br />
      Page not found.
    </div>
    <div
      class="button-home"
    >
      <a
        href="/home"
        rel="noreferrer"
        target="_self"
      >
        <img
          alt="Home icon"
          class="icon icon-home"
          src="image-mock"
        />
        <span
          class="text-home"
        >
          Home
        </span>
      </a>
    </div>
  </main>
</DocumentFragment>
`;
